tests/e2e/specs/out-of-game/authentication.spec.js

```diff
@@ -168,6 +168,13 @@ describe('Signing Up', () => {
     assertFailedAuth('/signup');
     assertSnackbar('That username is already registered to another user; try logging in!', 'error', 'auth');
   });
+  it('Rejects signup if username contains profanity', () => {
+    cy.get('[data-cy=username]').type('shitUser');
```

Comment:
Nice test. This string captures the key shortcoming of the implementation I previously suggested where we'd split on ' ' because there is no whitespace and not a direct member of our badWords, but  it's clearly profane. The same test would fail with that implementation which is a good sign for the test's usefulness and the correctness of your implementation